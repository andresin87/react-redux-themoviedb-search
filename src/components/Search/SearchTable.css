.SearchTable {
  padding: 5px;
  margin: 10px;
  border-radius: 8px;
}

@media (max-width: 1400px) {
  .SearchTableGenres {
    display: none;
  }
}

@media (max-width: 1100px) {
  .SearchTableReleaseDate {
    display: none;
  }
}

@media (max-width: 950px) {
  .SearchTablePopularity {
    display: none;
  }
}

@media (max-width: 700px) {
  .SearchTableVoteCount {
    display: none;
  }
}

@media (max-width: 550px) {
  .SearchTableVoteAverage {
    display: none;
  }
}
