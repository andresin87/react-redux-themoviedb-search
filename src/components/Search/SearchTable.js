import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Table } from 'antd';
import moment from 'moment';

import './SearchTable.css';
import { selectorSearch } from '../../reducers/search';
import { selectorGenres } from '../../reducers/genres';

const sort = (itemA, itemB) => {
  if (itemA > itemB) return 1;
  if (itemA < itemB) return -1;
  return 0;
};

class SearchTable extends PureComponent {
  static contextTypes = {
    router: PropTypes.object,
  };

  goToDetail(id, dataSource) {
    this.context.router.history.push({
      pathname: `/detail/${id}`,
      state: { dataSource },
    });
  }

  getColumns = () => {
    const { genresNames } = this.props;
    const columns = [
      {
        title: 'Title',
        dataIndex: 'title',
        key: 'title',
        className: 'SearchTableTitle',
        sorter: (a, b) => sort(a.title, b.title),
        width: 450,
      },
      {
        title: 'Vote average',
        dataIndex: 'vote_average',
        key: 'vote_average',
        className: 'SearchTableVoteAverage',
        sorter: (a, b) => sort(a.vote_average, b.vote_average),
        width: 150,
        align: 'center',
      },
      {
        title: 'Vote count',
        dataIndex: 'vote_count',
        key: 'vote_count',
        className: 'SearchTableVoteCount',
        sorter: (a, b) => sort(a.vote_count, b.vote_count),
        width: 150,
        align: 'center',
      },
      {
        title: 'Popularity',
        dataIndex: 'popularity',
        key: 'popularity',
        className: 'SearchTablePopularity',
        sorter: (a, b) => sort(a.popularity, b.popularity),
        width: 150,
        align: 'center',
      },
      {
        title: 'Release date',
        dataIndex: 'release_date',
        key: 'release_date',
        className: 'SearchTableReleaseDate',
        render: text => moment(text).format('DD/MM/YYYY'),
        sorter: (a, b) => sort(a.release_date, b.release_date),
        width: 150,
        align: 'center',
      },
      {
        title: 'Genres',
        dataIndex: 'genre_ids',
        key: 'genre_ids',
        render: ids =>
          ids
            .reduce((genresAsString, idGenre) => {
              return genresNames && genresNames[idGenre]
                ? `${genresAsString} ${genresNames[idGenre]},`
                : '';
            }, '')
            .slice(0, -1),
        className: 'SearchTableGenres',
      },
    ];
    return columns;
  };

  render() {
    const { isLoadingMovies, movies, isLoadingGenres } = this.props;
    const columns = this.getColumns();
    return (
      <div className="SearchTable">
        <Table
          loading={isLoadingMovies || isLoadingGenres}
          dataSource={movies}
          columns={columns}
          pagination={{ pageSize: 10 }}
          onRow={record => {
            return {
              onClick: () => this.goToDetail(record.id, movies),
            };
          }}
          rowKey={record => record.id}
        />
      </div>
    );
  }
}

SearchTable.propTypes = {
  movies: PropTypes.array.isRequired,
  isLoadingMovies: PropTypes.bool.isRequired,
  genresNames: PropTypes.object.isRequired,
  isLoadingGenres: PropTypes.bool.isRequired,
};

const mapStateToProps = state => {
  const { movies, isLoading: isLoadingMovies } = selectorSearch(state);
  const { names: genresNames, isLoading: isLoadingGenres } = selectorGenres(
    state
  );
  return {
    movies,
    isLoadingMovies,
    genresNames,
    isLoadingGenres,
  };
};

export default connect(mapStateToProps)(SearchTable);
