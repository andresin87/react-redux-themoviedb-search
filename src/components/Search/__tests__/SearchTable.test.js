import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TestUtils from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { Table } from 'antd';

import SearchTable from '../SearchTable';
import store from '../../../config/mockStore';

describe('SearchTable component', () => {
  const initialEntries = ['/'];
  const component = (
    <MemoryRouter initialEntries={initialEntries}>
      <Provider store={store}>
        <SearchTable />
      </Provider>
    </MemoryRouter>
  );

  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Snapshot matchs', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Check that on row click launchs an action', () => {
    const tree = TestUtils.renderIntoDocument(component);
    const table = TestUtils.findRenderedComponentWithType(tree, Table);
    const { onRow } = table.props;
    const id = 123;
    const record = { id };
    const onRowOnClickCallback = onRow(record).onClick;

    tree.history.push = jest.fn();
    onRowOnClickCallback();

    expect(tree.history.push).toBeCalled();
    expect(tree.history.push.mock.calls[0][0]).toEqual({
      pathname: '/detail/123',
      state: { dataSource: store.getState().search.movies },
    });
  });
});
