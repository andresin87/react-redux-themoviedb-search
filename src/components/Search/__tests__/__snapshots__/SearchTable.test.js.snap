// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchTable component Snapshot matchs 1`] = `
<div
  className="SearchTable"
>
  <code
    name="Antd Table"
    props={
      Object {
        "columns": Array [
          Object {
            "className": "SearchTableTitle",
            "dataIndex": "title",
            "key": "title",
            "sorter": [Function],
            "title": "Title",
            "width": 450,
          },
          Object {
            "align": "center",
            "className": "SearchTableVoteAverage",
            "dataIndex": "vote_average",
            "key": "vote_average",
            "sorter": [Function],
            "title": "Vote average",
            "width": 150,
          },
          Object {
            "align": "center",
            "className": "SearchTableVoteCount",
            "dataIndex": "vote_count",
            "key": "vote_count",
            "sorter": [Function],
            "title": "Vote count",
            "width": 150,
          },
          Object {
            "align": "center",
            "className": "SearchTablePopularity",
            "dataIndex": "popularity",
            "key": "popularity",
            "sorter": [Function],
            "title": "Popularity",
            "width": 150,
          },
          Object {
            "align": "center",
            "className": "SearchTableReleaseDate",
            "dataIndex": "release_date",
            "key": "release_date",
            "render": [Function],
            "sorter": [Function],
            "title": "Release date",
            "width": 150,
          },
          Object {
            "className": "SearchTableGenres",
            "dataIndex": "genre_ids",
            "key": "genre_ids",
            "render": [Function],
            "title": "Genres",
          },
        ],
        "dataSource": Array [
          Object {
            "adult": false,
            "backdrop_path": "/fgsHxz21B27hOOqQBiw9L6yWcM7.jpg",
            "genre_ids": Array [
              27,
              9648,
              53,
            ],
            "id": 439079,
            "original_language": "en",
            "original_title": "The Nun",
            "overview": "When a young nun at a cloistered abbey in Romania takes her own life, a priest with a haunted past and a novitiate on the threshold of her final vows are sent by the Vatican to investigate. Together they uncover the order’s unholy secret. Risking not only their lives but their faith and their very souls, they confront a malevolent force in the form of the same demonic nun that first terrorized audiences in “The Conjuring 2,” as the abbey becomes a horrific battleground between the living and the damned.",
            "popularity": 250.681,
            "poster_path": "/sFC1ElvoKGdHJIWRpNB3xWJ9lJA.jpg",
            "release_date": "2018-09-05",
            "title": "The Nun",
            "video": false,
            "vote_average": 6.1,
            "vote_count": 325,
          },
          Object {
            "adult": false,
            "backdrop_path": "/cI7oWsYbAhhzwgkAIOXprFzqJxO.jpg",
            "genre_ids": Array [
              27,
              878,
              28,
              35,
            ],
            "id": 346910,
            "original_language": "en",
            "original_title": "The Predator",
            "overview": "From the outer reaches of space to the small-town streets of suburbia, the hunt comes home. Now, the universe’s most lethal hunters are stronger, smarter and deadlier than ever before, having genetically upgraded themselves with DNA from other species. When a young boy accidentally triggers their return to Earth, only a ragtag crew of ex-soldiers and a disgruntled science teacher can prevent the end of the human race.",
            "popularity": 231.276,
            "poster_path": "/wMq9kQXTeQCHUZOG4fAe5cAxyUA.jpg",
            "release_date": "2018-09-13",
            "title": "The Predator",
            "video": false,
            "vote_average": 5.5,
            "vote_count": 108,
          },
          Object {
            "adult": false,
            "backdrop_path": "/lmZFxXgJE3vgrciwuDib0N8CfQo.jpg",
            "genre_ids": Array [
              12,
              878,
              28,
            ],
            "id": 299536,
            "original_language": "en",
            "original_title": "Avengers: Infinity War",
            "overview": "As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment - the fate of Earth and existence itself has never been more uncertain.",
            "popularity": 230.807,
            "poster_path": "/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg",
            "release_date": "2018-04-25",
            "title": "Avengers: Infinity War",
            "video": false,
            "vote_average": 8.3,
            "vote_count": 8004,
          },
          Object {
            "adult": false,
            "backdrop_path": "/96B1qMN9RxrAFu6uikwFhQ6N6J9.jpg",
            "genre_ids": Array [
              28,
              12,
              878,
            ],
            "id": 348350,
            "original_language": "en",
            "original_title": "Solo: A Star Wars Story",
            "overview": "Through a series of daring escapades deep within a dark and dangerous criminal underworld, Han Solo meets his mighty future copilot Chewbacca and encounters the notorious gambler Lando Calrissian.",
            "popularity": 174.43,
            "poster_path": "/4oD6VEccFkorEBTEDXtpLAaz0Rl.jpg",
            "release_date": "2018-05-15",
            "title": "Solo: A Star Wars Story",
            "video": false,
            "vote_average": 6.7,
            "vote_count": 1536,
          },
          Object {
            "adult": false,
            "backdrop_path": "/a9fRqsl6P27qxbDSb4pBXNv9Fwl.jpg",
            "genre_ids": Array [
              12,
              28,
              53,
            ],
            "id": 353081,
            "original_language": "en",
            "original_title": "Mission: Impossible - Fallout",
            "overview": "When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfill his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.",
            "popularity": 166.278,
            "poster_path": "/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg",
            "release_date": "2018-07-25",
            "title": "Mission: Impossible - Fallout",
            "video": false,
            "vote_average": 7.3,
            "vote_count": 1492,
          },
          Object {
            "adult": false,
            "backdrop_path": "/3s9O5af2xWKWR5JzP2iJZpZeQQg.jpg",
            "genre_ids": Array [
              28,
              12,
              878,
            ],
            "id": 351286,
            "original_language": "en",
            "original_title": "Jurassic World: Fallen Kingdom",
            "overview": "Three years after the demise of Jurassic World, a volcanic eruption threatens the remaining dinosaurs on the isla Nublar, so Claire Dearing, the former park manager, recruits Owen Grady to help prevent the extinction of the dinosaurs once again.",
            "popularity": 153.56,
            "poster_path": "/c9XxwwhPHdaImA2f1WEfEsbhaFB.jpg",
            "release_date": "2018-06-06",
            "title": "Jurassic World: Fallen Kingdom",
            "video": false,
            "vote_average": 6.5,
            "vote_count": 3291,
          },
          Object {
            "adult": false,
            "backdrop_path": "/rH79sB6Nkx4cMW3JzsUy7wK0rhX.jpg",
            "genre_ids": Array [
              28,
              878,
              53,
              27,
            ],
            "id": 345940,
            "original_language": "en",
            "original_title": "The Meg",
            "overview": "A deep sea submersible pilot revisits his past fears in the Mariana Trench, and accidentally unleashes the seventy foot ancestor of the Great White Shark believed to be extinct.",
            "popularity": 122.968,
            "poster_path": "/xqECHNvzbDL5I3iiOVUkVPJMSbc.jpg",
            "release_date": "2018-08-09",
            "title": "The Meg",
            "video": false,
            "vote_average": 6.1,
            "vote_count": 835,
          },
          Object {
            "adult": false,
            "backdrop_path": "/yCGEkEqy7uOo7NwlbgOzGrI8i43.jpg",
            "genre_ids": Array [
              28,
            ],
            "id": 135870,
            "original_language": "te",
            "original_title": "రెబెల్",
            "overview": "Rebel is a romantic action revenge drama, which revolves around the story of a happy-go-lucky guy. The hero (Prabhas) is the son of a good-natured mafia don (Krishnam Raju). The hero becomes Rebel after his father is killed. How he takes revenge upon the people, who killed his father, will form the second half of the movie. Apart from directing, Raghava Lawrence also handles the choreography, screenplay and music direction for the film. The film is being produced by J. Bhaghawan and J. Pulla Rao under Sri Balaji Cine Media banner.",
            "popularity": 122.914,
            "poster_path": "/dNTzorrVmcJ25fUGzs9FfkoXb0P.jpg",
            "release_date": "2012-09-28",
            "title": "Rebel",
            "video": false,
            "vote_average": 4.2,
            "vote_count": 20,
          },
          Object {
            "adult": false,
            "backdrop_path": "/3P52oz9HPQWxcwHOwxtyrVV1LKi.jpg",
            "genre_ids": Array [
              28,
              35,
              878,
            ],
            "id": 383498,
            "original_language": "en",
            "original_title": "Deadpool 2",
            "overview": "Wisecracking mercenary Deadpool battles the evil and powerful Cable and other bad guys to save a boy's life.",
            "popularity": 118.424,
            "poster_path": "/to0spRl1CMDvyUbOnbb4fTk3VAd.jpg",
            "release_date": "2018-05-15",
            "title": "Deadpool 2",
            "video": false,
            "vote_average": 7.5,
            "vote_count": 4893,
          },
          Object {
            "adult": false,
            "backdrop_path": "/uKwOX7MtKlAaGeCQe6c4jc1vZpj.jpg",
            "genre_ids": Array [
              28,
              12,
              14,
              16,
            ],
            "id": 507569,
            "original_language": "ja",
            "original_title": "劇場版 七つの大罪 天空の囚われ人",
            "overview": "The Seven Deadly Sins travel to a remote land in search of the phantom ingredient \\"sky fish.\\" Meliodas and Hawk end up at a \\"Sky Palace\\" that exists above the clouds, where all the residents have wings. Meliodas is mistaken for a boy who committed a crime and is thrown in prison. Meanwhile, the residents are preparing a ceremony for defense against a ferocious beast that awakens once every 3,000 years. But the Six Knights of Black, a Demon Clan army, arrives and removes the seal on the beast in order to threaten the lives of the residents of Sky Palace. Meliodas and his allies meet the Six Knights of Black in battle.",
            "popularity": 111.455,
            "poster_path": "/r6pPUVUKU5eIpYj4oEzidk5ZibB.jpg",
            "release_date": "2018-08-18",
            "title": "The Seven Deadly Sins: Prisoners of the Sky",
            "video": false,
            "vote_average": 5.2,
            "vote_count": 23,
          },
          Object {
            "adult": false,
            "backdrop_path": "/b6ZJZHUdMEFECvGiDpJjlfUWela.jpg",
            "genre_ids": Array [
              28,
              12,
              14,
              878,
            ],
            "id": 284054,
            "original_language": "en",
            "original_title": "Black Panther",
            "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without. Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister, members of the Dora Milaje (the Wakandan 'special forces') and an American secret agent, to prevent Wakanda from being dragged into a world war.",
            "popularity": 100.458,
            "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
            "release_date": "2018-02-13",
            "title": "Black Panther",
            "video": false,
            "vote_average": 7.4,
            "vote_count": 8065,
          },
          Object {
            "adult": false,
            "backdrop_path": "/scQf03Fm3jeyv4FH04qvi4fp4wh.jpg",
            "genre_ids": Array [
              80,
              35,
              28,
              53,
            ],
            "id": 402900,
            "original_language": "en",
            "original_title": "Ocean's Eight",
            "overview": "Debbie Ocean, a criminal mastermind, gathers a crew of female thieves to pull off the heist of the century at New York's annual Met Gala.",
            "popularity": 87.901,
            "poster_path": "/MvYpKlpFukTivnlBhizGbkAe3v.jpg",
            "release_date": "2018-06-07",
            "title": "Ocean's Eight",
            "video": false,
            "vote_average": 7,
            "vote_count": 1614,
          },
          Object {
            "adult": false,
            "backdrop_path": "/4MJkbiDU5lr6TAwSLPV14C9w0vs.jpg",
            "genre_ids": Array [
              10749,
            ],
            "id": 390845,
            "original_language": "ko",
            "original_title": "맛있는 비행",
            "overview": "The in- flight service for Hong Kong starts now!  An erotic story that took place in the spacious blue sky!  An innocent actress runs away from the scandal with an idol, the impudent idol that ruined her career, a passionate manager who is devoted to her and his ex-girlfriend who is now a sexy stewardess. These four that should never be together run into each other in an airplane. No one can run away and the most cheeky and erotic things happen here.",
            "popularity": 79.05,
            "poster_path": "/kGlBKJBosHTA4elgHS54olzD4oS.jpg",
            "release_date": "2015-11-01",
            "title": "A Delicious Flight",
            "video": false,
            "vote_average": 3.7,
            "vote_count": 24,
          },
          Object {
            "adult": false,
            "backdrop_path": "/7UYdlNrL0mI9MzoqSyCvN3PJrvH.jpg",
            "genre_ids": Array [
              28,
              14,
              27,
            ],
            "id": 460885,
            "original_language": "en",
            "original_title": "Mandy",
            "overview": "The Shadow Mountains, 1983. Red and Mandy lead a loving and peaceful existence; but when their pine-scented haven is savagely destroyed, Red is catapulted into a phantasmagoric journey filled with bloody vengeance and laced with fire.",
            "popularity": 70.225,
            "poster_path": "/m0yf7J7HsKeK6E81SMRcX8vx6mH.jpg",
            "release_date": "2018-05-12",
            "title": "Mandy",
            "video": false,
            "vote_average": 7.4,
            "vote_count": 23,
          },
          Object {
            "adult": false,
            "backdrop_path": "/oMKFQmoVgB69fyXfSMu0lGlHJP2.jpg",
            "genre_ids": Array [
              28,
              53,
              18,
            ],
            "id": 447200,
            "original_language": "en",
            "original_title": "Skyscraper",
            "overview": "Framed and on the run, a former FBI agent must save his family from a blazing fire in the world's tallest building.",
            "popularity": 69.816,
            "poster_path": "/5LYSsOPzuP13201qSzMjNxi8FxN.jpg",
            "release_date": "2018-07-11",
            "title": "Skyscraper",
            "video": false,
            "vote_average": 6,
            "vote_count": 670,
          },
          Object {
            "adult": false,
            "backdrop_path": "/1sZ9Nnic1ldHhHttAMDmNxaNM04.jpg",
            "genre_ids": Array [
              878,
              27,
              53,
            ],
            "id": 158015,
            "original_language": "en",
            "original_title": "The Purge",
            "overview": "Given the country's overcrowded prisons, the U.S. government begins to allow 12-hour periods of time in which all illegal activity is legal. During one of these free-for-alls, a family must protect themselves from a home invasion.",
            "popularity": 68.603,
            "poster_path": "/tGGJOuLHX7UDlTz57sjfhW1qreP.jpg",
            "release_date": "2013-05-31",
            "title": "The Purge",
            "video": false,
            "vote_average": 6.1,
            "vote_count": 3819,
          },
          Object {
            "adult": false,
            "backdrop_path": "/zWGAnbxjjwY3xFGuOeR26LGbBlG.jpg",
            "genre_ids": Array [
              27,
              53,
            ],
            "id": 238636,
            "original_language": "en",
            "original_title": "The Purge: Anarchy",
            "overview": "One night per year, the government sanctions a 12-hour period in which citizens can commit any crime they wish -- including murder -- without fear of punishment or imprisonment. Leo, a sergeant who lost his son, plans a vigilante mission of revenge during the mayhem. However, instead of a death-dealing avenger, he becomes the unexpected protector of four innocent strangers who desperately need his help if they are to survive the night.",
            "popularity": 67.961,
            "poster_path": "/l1DRl40x2OWUoPP42v8fjKdS1Z3.jpg",
            "release_date": "2014-07-17",
            "title": "The Purge: Anarchy",
            "video": false,
            "vote_average": 6.6,
            "vote_count": 3048,
          },
          Object {
            "adult": false,
            "backdrop_path": "/fdFIVBpq57ZbHok07SI3gNLWUbO.jpg",
            "genre_ids": Array [
              35,
              18,
              10749,
            ],
            "id": 462919,
            "original_language": "en",
            "original_title": "Sierra Burgess Is a Loser",
            "overview": "A case of mistaken identity results in unexpected romance when the most popular girl in high school and the biggest loser must come together to win over their crushes.",
            "popularity": 66.973,
            "poster_path": "/vjnHRySiVM4YcBYzjCxw344y9g.jpg",
            "release_date": "2018-09-07",
            "title": "Sierra Burgess Is a Loser",
            "video": false,
            "vote_average": 7.2,
            "vote_count": 604,
          },
          Object {
            "adult": false,
            "backdrop_path": "/kaIfm5ryEOwYg8mLbq8HkPuM1Fo.jpg",
            "genre_ids": Array [
              28,
              12,
            ],
            "id": 284053,
            "original_language": "en",
            "original_title": "Thor: Ragnarok",
            "overview": "Thor is on the other side of the universe and finds himself in a race against time to get back to Asgard to stop Ragnarok, the prophecy of destruction to his homeworld and the end of Asgardian civilization, at the hands of an all-powerful new threat, the ruthless Hela.",
            "popularity": 66.111,
            "poster_path": "/rzRwTcFvttcN1ZpX2xv4j3tSdJu.jpg",
            "release_date": "2017-10-25",
            "title": "Thor: Ragnarok",
            "video": false,
            "vote_average": 7.5,
            "vote_count": 8278,
          },
          Object {
            "adult": false,
            "backdrop_path": "/dnaitaoCh8MftfYEVnprcuYExZp.jpg",
            "genre_ids": Array [
              28,
              27,
              878,
              53,
            ],
            "id": 442249,
            "original_language": "en",
            "original_title": "The First Purge",
            "overview": "To push the crime rate below one percent for the rest of the year, the New Founding Fathers of America test a sociological theory that vents aggression for one night in one isolated community. But when the violence of oppressors meets the rage of the others, the contagion will explode from the trial-city borders and spread across the nation.",
            "popularity": 63.531,
            "poster_path": "/2slvblTroiT1lY9bYLK7Amigo1k.jpg",
            "release_date": "2018-07-04",
            "title": "The First Purge",
            "video": false,
            "vote_average": 6,
            "vote_count": 649,
          },
        ],
        "loading": false,
        "onRow": [Function],
        "pagination": Object {
          "pageSize": 10,
        },
        "rowKey": [Function],
      }
    }
  />
</div>
`;
